File: 프로그래머스/1/12901. 2016년/2016년.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int a, int b) {
    vector<string> week = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};
    vector<int> month = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int n = 0;
    for(int i = 0; i < a - 1; i++){
        n += month[i];
    }
    return week[(n + b + 4) % 7];
}

File: 프로그래머스/1/12901. 2016년/2016년.js
function solution(a, b) {
    let week = ["SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"];
    let month = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    let n = 0;
    for(let i = 0; i < a - 1; i++){
        n += month[i];
    }
    return week[(n + b + 4) % 7];
}